x-pack/plugins/fleet/server/services/actions/utils.test.ts

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as esKuery from '@kbn/es-query';
+
+import {
+  validateFilterKueryNode,
+  allowedFleetActionsFields,
+  allowedFleetActionsResultsFields,
+  isFieldDefined,
+  hasFieldKeyError,
+  type IndexType,
+} from './utils';
+
+describe('utils', () => {
+  describe('#validateFilterKueryNode', () => {
+    it('should accept only allowed search fields', () => {
+      const validationObject = validateFilterKueryNode({
+        astFilter: esKuery.fromKueryExpression(
+          'action_id: "1" or action_id: "2" and (input_type: "endpoint" and @timestamp <= "now" and @timestamp >= "now-2d")'
+        ),
+        types: ['keyword', 'date'],
+        indexMapping: allowedFleetActionsFields,
+      });
+
+      expect(validationObject).toEqual([
+        {
+          astPath: 'arguments.0',
+          error: null,
+          key: 'action_id',
+          type: 'keyword',
+        },
+        {
+          astPath: 'arguments.1.arguments.0',
+          error: null,
+          key: 'action_id',
+          type: 'keyword',
+        },
+        {
+          astPath: 'arguments.1.arguments.1.arguments.0',
+          error: null,
+          key: 'input_type',
+          type: 'keyword',
+        },
+        {
+          astPath: 'arguments.1.arguments.1.arguments.1',
+          error: null,
+          key: '@timestamp',
+          type: 'date',
+        },
+        {
+          astPath: 'arguments.1.arguments.1.arguments.2',
+          error: null,
+          key: '@timestamp',
+          type: 'date',
+        },
+      ]);
+    });
+
+    it('should not accept if any search fields are not allowed', () => {
+      const validationObject = validateFilterKueryNode({
+        astFilter: esKuery.fromKueryExpression(
+          'action_id: "1" and expiration: "2023-06-21T10:55:36.481Z" and (input_type: "endpoint" and @timestamp <= "now" and @timestamp >= "now-2d")'
+        ),
+        types: ['keyword', 'date'],
+        indexMapping: allowedFleetActionsFields,
+      });
+
+      expect(validationObject).toEqual([
+        {
+          astPath: 'arguments.0',
+          error: null,
+          key: 'action_id',
+          type: 'keyword',
+        },
+        {
+          astPath: 'arguments.1',
+          error:
+            "This key 'expiration' does NOT exist in [action_id,agents,input_type,@timestamp,type,user_id] .fleet-actions* index patterns",
```

Comment:
nit: I think we shouldn't list all existing fields in the error message. Instead the error message would be clearer to say something like `The field 'expiration' does not exist in .fleet-actions index mappings`

Reply:
done 19e31f859bf96727eb2d3d5b25b014044636c067